src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,341 @@
+import { PostPerformanceData } from './post-performance-data';
+
+// Number of attempts to fetch the data before displaying an error.
+const FETCH_RETRIES = 3;
+
+/**
+ * Specifies the form of component props.
+ */
+interface PostDetailsSectionProps {
+	data: PostPerformanceData;
+}
+
+/**
+ * Outputs the current post's details or shows an error message on failure.
+ */
+function CurrentPostDetails() {
+	const [ loading, setLoading ] = useState<boolean>( true );
+	const [ error, setError ] = useState( null );
+	const [ postDetailsData, setPostDetails ] = useState<PostPerformanceData>( null );
+	const provider = new CurrentPostDetailsProvider();
+
+	useEffect( () => {
+		const fetchPosts = async ( retries: number ) => {
+			provider.getCurrentPostDetails()
+				.then( ( result ) => {
+					setPostDetails( result );
+					setLoading( false );
+				} )
+				.catch( async ( err ) => {
+					if ( retries > 0 ) {
+						await new Promise( ( r ) => setTimeout( r, 500 ) );
+						await fetchPosts( retries - 1 );
+					} else {
+						setError( err );
+						setLoading( false );
+					}
+				} );
+		};
+
+		setLoading( true );
+		fetchPosts( FETCH_RETRIES );
+	}, [] );
+
+	if ( error ) {
+		// "Soft" error for which we do not want to show an "Error:" prefix.
+		if ( 'string' === typeof error ) {
+			return <p>{ error }</p>;
+		}
+
+		// Error coming from apiFetch.
+		if ( error?.message ) {
+			return <p>{ __( 'Error:', 'wp-parsely' ) } { error.message }</p>;
+		}
+
+		// Error coming from the WordPress REST API.
+		const errorMessage = JSON.stringify( error ).match( /\[\"(.*?)\"\]/ )[ 1 ];
+		return <p>{ __( 'Error:', 'wp-parsely' ) } { errorMessage }</p>;
+	}
+
+	return (
+		<>
+			{ loading ? <Spinner /> : <CurrentPostDetailsSections data={ postDetailsData } /> }
+		</>
+	);
+}
+
+/**
+ * Outputs all the "Current Post Details" sections.
+ *
+ * @param {PostDetailsSectionProps} props Props needed to populate the sections.
+ */
+function CurrentPostDetailsSections( props: PostDetailsSectionProps ) {
+	const data = props.data;
+
+	return (
+		<div className="current-post-details-panel">
+			{ DataPeriodSection( data ) }
+			{ GeneralPerformanceSection( data ) }
+			{ ReferrerTypesSection( data ) }
+			{ TopReferrersSection( data ) }
+			{ ActionsSection( data ) }
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Period" section, which denotes the period for which data is
+ * shown.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function DataPeriodSection( data: PostPerformanceData ) {
+	const period = data.period;
+
+	// Get the date (in short format) on which the period starts.
+	const periodStart = Intl.DateTimeFormat(
+		document.documentElement.lang,
+		{ month: 'short', day: 'numeric' }
+	).format( new Date( period.start ) );
+
+	return (
+		<div className="section period">
+			{
+				/* translators: Number of days for which data is being shown */
+				sprintf( __( 'Last %d Days', 'wp-parsely' ), period.days )
```

Comment:
Even in English, this string is going to be wrong if the number of days is exactly 1. 

Use https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/#n to allow singular and plural strings, and pass in the `period.days`, and let the i18n system choose the right string for the right number.

Reply:
We currently get the data for 7 days in a hardcoded manner, so as things stand it cannot be 1.

As such, I won't proceed with that change, but your point is clear and correct. Given we ever allow 1 to be an option, we will revise this.

Reply:
If the data is always 7 days, why is there even a `period.days` variable?

Reply:
It caters for the possibility of adding multiple predefined periods in a future iteration of the feature. This is why we also have the `setDataPeriod()` function for example.